modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/api/SessionImpl.java

```diff
@@ -149,7 +151,13 @@ public int defaultPageSize() {
     /** {@inheritDoc} */
     @Override
     public SessionBuilder toBuilder() {
-        return new SessionBuilderImpl(qryProc, new HashMap<>(props.toMap()))
+        Map<String, Object> propertyMap = new HashMap<>();
+
+        for (Map.Entry<Property<?>, Object> entry : props) {
+            propertyMap.put(entry.getKey().name, entry.getValue());
+        }
+
+        return new SessionBuilderImpl(qryProc, propertyMap)
```

Comment:
```suggestion      
        return new SessionBuilderImpl(qryProc, props.toMap())
```
Actually, PropertiesHolder API looks ambiguous.
It is not clean if toMap() return a copy of properties, or just adapter or internal map.
I'd expect toMap() returns a copy of props, and it make sense to add asMap() that will avoid copying.

Reply:
If we clam PropertiesHolder is immutable, and can be changes only via copying with builder,
then maybe defensive-copying can be ommited or replaced with Collections.unmodifiableMap() or Map.ofEntries().